dataframely/_base_schema.py

```diff
@@ -137,6 +137,29 @@ def __new__(
                         f"which are not in the schema: {missing_list}."
                     )
 
+        # 3) Check that all members are non-pathological (i.e., user errors).
```

Comment:
One question: why are we asserting that types are specific disallowed types instead of checking that they are _allowed_. It seems like we'd be missing other invalid types here 🤔

Reply:
Before this PR, we have already been allowing users to have arbitrary extra members. To not break this, we decided to just raise explicitly for stuff where we are 100% sure that the user cannot have intended what they did.